### DISCUSSION TITLE
Performance - Support Next.js App Router

### DISCUSSION THREAD

**Discussion by sethdaily** [tinacms/tinacms#4479] [Ideas]
Pain: New Next.js projects will always want to use the App Router...but TinaCMS only supports the Pages Router (legacy, and a lot slower).

Therefore, you are either forced to use the Pages Router (legacy), or you can't use TinaCMS.

Solution: Support Next.js App Router

**Comment by arzafran** [comment]
it does work with app router, examples may be outdated but we've used it in production and works flawlessly.

**Reply by jeremyblack** [reply]
Can you elaborate on the expected (or a sample) folder structure when using Tina with app router.

**Reply by Harry-Ross** [reply]
Sorry for the confusion - we're aware that there is a way to change between a client-side component and server-side components using draft mode, but what we are proposing is native RSC support in TinaCMS (i.e. what Sanity and Builder.IO have).
 
@jeremyblack see here for an example on how to use the current version of the React plugin with App router :) https://github.com/twittkopf/tinacms-visual-editing-nextjs-app-dir/blob/main/src/app/demo/blog/%5Bfilename%5D/page.js
